### PR TITLE
[Spring] APM and CA Certificate Supports Shareable Build Result

### PR DESCRIPTION
**Pull request by yilims**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/spring/azext_spring/_validators_enterprise.py [hunk 1]
```diff
@@ -118,17 +120,6 @@ def validate_build_service(namespace):
...
-                and namespace.disable_build_service:
-            raise InvalidArgumentValueError(
-                "Conflict detected: '--registry-server', '--registry-username' and '--registry-password' "
-                "can not be set with '--disable-build-service'.")
-        if namespace.disable_build_service:
-            namespace.disable_app_insights = True
-            if namespace.app_insights or namespace.app_insights_key:
-                raise InvalidArgumentValueError(
-                    "Conflict detected: '--app-insights' or '--app-insights-key' "
-                    "can not be set with '--disable-build-service'.")
```
Comment (smile37773) [inline src/spring/azext_spring/_validators_enterprise.py:131]: why do we delete these lines？

Comment (yilims) [reply src/spring/azext_spring/_validators_enterprise.py:131]: Previously App Insights is disabled when build service is diabled as APM is not ready. Now App Insights can be enabled when build service is diabled as APM is ready

Comment (smile37773) [reply src/spring/azext_spring/_validators_enterprise.py:131]: OK, I think you should keep lines 121-125, which are not related to APMs.

Comment (yilims) [reply src/spring/azext_spring/_validators_enterprise.py:131]: restored lines 121-125

**Review by zhoxing-ms — APPROVED** [review]

**Review by smile37773 — APPROVED** [review]

**Pull request merged**